openapi.yaml

```diff
@@ -13110,6 +13248,158 @@ components:
+            When this Firewall was last updated.
+          example: '2018-01-02T00:01:01'
+          x-linode-cli-display: 5
+        status:
+          type: string
+          readOnly: true
+          description: >
+           The status of this Firewall service.
+          enum:
+          - enabled
+          - disabled
+          - deleted
+          example: enabled
+          x-linode-cli-display: 3
+        rules:
+          type: object
+          description: >
+             The inbound and outbound access rules to apply to the Firewall.
+
+              * A minimum of one open inbound port is required. Any inbound
+                port that is not open will be blocked.
+              * Outbound rules are optional. When no outbound ports are open, all outbound
+                ports are allowed. If you open one outbound port or more, all
+                remaining outbound ports will be blocked.
+          properties:
+            inbound:
+              $ref: '#/components/schemas/FirewallRuleConfig'
+            outbound:
+              $ref: '#/components/schemas/FirewallRuleConfig'
+          required:
+            - inbound
+        tags:
+          x-linode-filterable: true
+          description: >
+            An array of tags applied to this object. Tags are for organizational
+            purposes only.
+          type: array
+          items:
+            type: string
+          example:
+          - example tag
+          - another example
+        devices:
+          type: object
+          description: >
+            The devices that this Firewall will be applied to. Currently, you
+            can only add a Firewall to Linodes.
+          properties:
+            linodes:
+              description: An array of Linode ids
+              type: array
+              items:
+                type: integer
+              example:
+                - 123
+                - 456
+              x-linode-cli-display: 6
+    FirewallRuleConfig:
+      type: object
+      description: >
+        This Firewall's access rules. To open traffic on a range of ports
+        specify a `start_port` and an `end_port`.
+      required:
+      - protocol
+      - start_port
+      properties:
+        protocol:
+          type: string
+          enum:
+          - ALL
+          - TCP
+          - UDP
+          - ICMP
+          description: >
+            The type of network traffic to allow.
+          example: TCP
+        start_port:
+          type: integer
+          description: >
+            The port on which traffic will be allowed. Must be a value of `1-65535`.
+          example: 20
+        end_port:
+          type: integer
+          description: >
+            The end port for a range of ports on which traffic will be allowed. This port must be greater than or equal to the `start_port`
+            and must be a value of `1-65535`.
+          example: 22
+        addresses:
+          type: object
+          description: >
```

Comment:
Can we add a note about the "all traffic" network masks here? 0.0.0.0/0 for ipv4 and ::0/0 for ipv6